fix: loading utf-8 file when locale encoding is set to ascii

taimoorzaeem opened the pull request:
Configurator-pg fails to load UTF-8 encoded files when the locale settings are set to ASCII chars. This commit fixes this by using a safer and faster UTF-8 decoding when reading files.

See: https://github.com/PostgREST/postgrest/issues/4386#issuecomment-3377862644 for more info on this.

steve-chavez commented on tests/resources/err-import.cfg.err:
```diff
@@ -1 +1 @@
-tests/resources/not-exist.cfg: openFile: does not exist (No such file or directory)
\ No newline at end of file
+tests/resources/not-exist.cfg: withBinaryFile: does not exist (No such file or directory)
```
So this changes because we now use the ByteString.readFile function: https://hackage-content.haskell.org/package/bytestring-0.12.2.0/docs/Data-ByteString.html#v:readFile

steve-chavez commented on tests/Test.hs:
```diff
@@ -20,6 +22,7 @@ tests :: [Test]
 tests =
     [ testCase "read-simple" $ readTest "simple.cfg"
     , testCase "read-pathological" $ readTest "pathological.cfg"
+    , testCase "read-utf-8-with-ascii-locale" $ readTestWithLocale "utf-8.cfg" E.ascii
```
Q: Prior to the change, did this one fail with:

```
hGetContents: invalid argument (cannot decode byte sequence starting from ..
```

Same as https://github.com/PostgREST/postgrest/issues/4379#issue-3481347325?

taimoorzaeem replied:
Yes, exactly! If I remove this change and re-run tests, I get:

```
read-utf-8-with-ascii-locale: [Failed]
ERROR: tests/resources/utf-8.cfg: hGetContents: invalid argument (cannot decode byte sequence starting from 195)
```

EDIT: A couple other tests slightly change in output, so I modified accordingly.

taimoorzaeem committed:
fix: loading utf-8 file when locale encoding is set to ascii

Configurator-pg fails to load UTF-8 encoded files when the locale
settings are set to ASCII chars. This commit fixes this by using a
safer and faster UTF-8 decoding when reading files.

See: https://github.com/PostgREST/postgrest/issues/4386#issuecomment-3377862644
for more info on this.

Signed-off-by: Taimoor Zaeem <taimoorzaeem@gmail.com>

steve-chavez reviewed (approved):
Awesome work :1st_place_medal:
